Framework/Kernel/src/Material.cpp

```diff
@@ -137,15 +476,63 @@ void Material::saveNexus(::NeXus::File *file, const std::string &group) const {
 void Material::loadNexus(::NeXus::File *file, const std::string &group) {
   file->openGroup(group, "NXdata");
   file->getAttr("name", m_name);
+  int version;
+  file->getAttr("version", version);
 
-  // Find the element
-  uint16_t element_Z, element_A;
-  file->readData("element_Z", element_Z);
-  file->readData("element_A", element_A);
-  try {
-    m_element = Mantid::PhysicalConstants::getNeutronAtom(element_Z, element_A);
-  } catch (std::runtime_error &) { /* ignore and use the default */
+  if (version == 1) {
+    // Find the element
+    uint16_t element_Z, element_A;
+    file->readData("element_Z", element_Z);
+    file->readData("element_A", element_A);
+    try {
+      m_chemicalFormula.clear();
+      if (element_Z > 0) {
+        FormulaUnit formulaUnit{
+            boost::make_shared<Atom>(
+                Mantid::PhysicalConstants::getAtom(element_Z, element_A)),
+            1.};
+        m_chemicalFormula.push_back(formulaUnit);
+      } else {
+        FormulaUnit formulaUnit{
+            boost::make_shared<Atom>(Mantid::PhysicalConstants::getNeutronAtom(
+                element_Z, element_A)),
+            1.};
+        m_chemicalFormula.push_back(formulaUnit);
+      }
+    } catch (std::runtime_error &) { /* ignore and use the default */
+    }
+  } else if (version == 2) {
+    std::string style;
+    file->getAttr("formulaStyle", style);
+
+    if (style == "formula") {
+      std::string formula;
+      file->readData("chemical_formula", formula);
+      this->m_chemicalFormula = Material::parseChemicalFormula(formula);
+      this->countAtoms();
+    } else if (style == "userdefined") {
+      NeutronAtom neutron;
+      file->readData("coh_scatt_length_real", neutron.coh_scatt_length_real);
+      file->readData("coh_scatt_length_img", neutron.coh_scatt_length_img);
+      file->readData("inc_scatt_length_real", neutron.inc_scatt_length_real);
+      file->readData("inc_scatt_length_img", neutron.inc_scatt_length_img);
+      file->readData("coh_scatt_xs", neutron.coh_scatt_xs);
+      file->readData("inc_scatt_xs", neutron.inc_scatt_xs);
+      file->readData("tot_scatt_xs", neutron.tot_scatt_xs);
+      file->readData("abs_scatt_xs", neutron.abs_scatt_xs);
+      file->readData("tot_scatt_length", neutron.tot_scatt_length);
+      file->readData("coh_scatt_length", neutron.coh_scatt_length);
+      file->readData("inc_scatt_length", neutron.inc_scatt_length);
+
+      FormulaUnit formulaUnit{boost::make_shared<Atom>(neutron), 1.};
+      m_chemicalFormula.push_back(formulaUnit);
```

Comment:
We could eliminate the temporary using `emplace_back` if we added a constructor to Formula unit.